packages/sanity/src/structure/panes/document/documentPanel/banners/OpenReleaseToEditBanner.tsx

```diff
@@ -0,0 +1,123 @@
+import {Flex, Text} from '@sanity/ui'
+import {useCallback, useMemo} from 'react'
+import {
+  getReleaseIdFromReleaseDocumentId,
+  getReleaseTone,
+  getVersionFromId,
+  isVersionId,
+  Translate,
+  useActiveReleases,
+  useDocumentVersions,
+  useOnlyHasVersions,
+  useSetPerspective,
+  useTranslation,
+  VersionInlineBadge,
+} from 'sanity'
+import {structureLocaleNamespace} from 'sanity/structure'
+
+import {Button} from '../../../../../ui-components'
+import {Banner} from './Banner'
+
+export function OpenReleaseToEditBanner({
+  documentId,
+  isPinnedDraftOrPublished,
+}: {
+  documentId: string
+  isPinnedDraftOrPublished: boolean
+}): React.JSX.Element | null {
+  const {onlyHasVersions} = useOnlyHasVersions({documentId})
+
+  /** this banner should only be rendered in specific cases
+   * 1. when the document is not a draft or published
+   * 2. when the document only has version or versions
+   * 3. when the global perspective is not set (published or draft)
+   */
+
+  if (!isVersionId(documentId)) {
+    return null
+  }
+
+  if (!onlyHasVersions || !isPinnedDraftOrPublished) {
+    return null
+  }
+
+  return <OpenReleaseToEditBannerInner documentId={documentId} />
+}
+
+export function OpenReleaseToEditBannerInner({
+  documentId,
+}: {
+  documentId: string
+}): React.JSX.Element {
+  const {data: activeReleases} = useActiveReleases()
+  const setPerspective = useSetPerspective()
+  const releaseId = getVersionFromId(documentId) ?? ''
+  const currentVersion = useMemo(
+    () => activeReleases.find((version) => version._id.includes(releaseId)),
+    [activeReleases, releaseId],
+  )
+  const {data: documentVersions} = useDocumentVersions({documentId})
+
+  const documentVersionsTitleList = useMemo(
+    () =>
+      activeReleases
+        .filter((version) => {
+          return documentVersions.find((release) => {
+            const r = getVersionFromId(release) ?? ''
+            return getReleaseIdFromReleaseDocumentId(version._id) === r
+          })
+        })
+        .map((version) => version.metadata.title),
+    [activeReleases, documentVersions],
+  )
+  const tone = currentVersion && getReleaseTone(currentVersion)
+  const {t} = useTranslation(structureLocaleNamespace)
+
+  const handleGoToEdit = useCallback(async () => {
+    setPerspective(releaseId)
+  }, [releaseId, setPerspective])
+
+  return (
+    <Banner
+      tone={tone}
+      paddingY={0}
+      data-testId="open-release-to-edit-banner"
```

Comment:
I think this should be `data-testid`. React is logging an error to the browser console when it renders:

```
React does not recognize the `data-testId` prop on a DOM element
```

```suggestion
      data-testid="open-release-to-edit-banner"
```

Reply:
good catch!